javaScript/signup.js

```diff
@@ -0,0 +1,134 @@
+  const emailInput = document.getElementById("email");
+  const emailError = document.getElementById("email-error");
+
+  const usernameInput = document.getElementById("username");
+  const usernameError = document.getElementById("username-error");
+
+  const passwordInput = document.getElementById("password");
+  const passwordError = document.getElementById("password-error");
+
+  const confirmPasswordInput = document.getElementById("confirm-password");
+  const confirmPasswordError = document.getElementById("confirm-password-error");
+
+  const passwordToggle = document.querySelector("#password-toggle");
+  const confirmPasswordToggle = document.querySelector("#confirm-password-toggle");
+
+  const signupButton = document.querySelector(".login-btn");
+
+  function showError(input, errorElement, message) {
+    input.style.border = "2px solid red";
+    errorElement.textContent = message;
+    errorElement.style.color = "red";
+    errorElement.style.display = "block";
+    updateSignupButtonState();
+  }
+
+  function hideError(input, errorElement) {
+    input.style.border = "";
+    errorElement.textContent = "";
+    errorElement.style.display = "none";
+    updateSignupButtonState();
+  }
+
+  function validateInput() {
+    let isValid = true;
+
+    if (emailInput.value.trim() === "" || emailError.textContent !== "") isValid = false;
+    if (usernameInput.value.trim() === "" || usernameError.textContent !== "") isValid = false;
+    if (passwordInput.value.trim() === "" || passwordError.textContent !== "") isValid = false;
+    if (confirmPasswordInput.value.trim() === "" || confirmPasswordError.textContent !== "") isValid = false;
+
+    return isValid;
+  }
+
+  function updateSignupButtonState() {
+    signupButton.disabled = !validateInput();
+  }
+
+  emailInput.addEventListener("blur", function () {
+    const emailValue = emailInput.value.trim();
+    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    if (emailValue === "") {
+      showError(emailInput, emailError, "이메일을 입력해주세요.");
+    } else if (!emailPattern.test(emailValue)) {
+      showError(emailInput, emailError, "잘못된 이메일 형식입니다.");
+    } else {
+      hideError(emailInput, emailError);
+    }
+  });
+
+  usernameInput.addEventListener("blur", function () {
+    if (usernameInput.value.trim() === "") {
+      showError(usernameInput, usernameError, "닉네임을 입력해주세요.");
+    } else {
+      hideError(usernameInput, usernameError);
+    }
+  });
+
+  passwordInput.addEventListener("blur", function () {
+    const passwordValue = passwordInput.value.trim();
+
+    if (passwordValue === "") {
+      showError(passwordInput, passwordError, "비밀번호를 입력해주세요.");
+    } else if (passwordValue.length < 8) {
+      showError(passwordInput, passwordError, "비밀번호를 8자 이상 입력해주세요.");
+    } else {
+      hideError(passwordInput, passwordError);
+    }
+  });
+
+  passwordInput.addEventListener("input", function () {
+    validateConfirmPassword();
+  });
```

Comment:
💊 제안
이벤트 리스너에서 콜백 함수를 전달할 때, 익명 함수로 감싸지 않고 바로 함수 참조를 전달하는 것이 더 간결하고 효율적일 것 같습니다~

```suggestion
  passwordInput.addEventListener("input", validateConfirmPassword);
```